src/NegRiskAdapterMapping.ts

```diff
@@ -0,0 +1,124 @@
+import { BigInt, ByteArray } from '@graphprotocol/graph-ts';
+
+import {
+  PositionSplit,
+  PositionsMerge,
+  PositionsConverted,
+  MarketPrepared,
+  QuestionPrepared,
+  PayoutRedemption,
+} from './types/NegRiskAdapter/NegRiskAdapter';
+import {
+  Merge,
+  Split,
+  Redemption,
+  NegRiskConversion,
+  NegRiskEvent,
+} from './types/schema';
+import { markAccountAsSeen, requireAccount } from './utils/account-utils';
+
+export function handlePositionSplit(event: PositionSplit): void {
+  requireAccount(event.params.stakeholder.toHexString(), event.block.timestamp);
+  markAccountAsSeen(
+    event.params.stakeholder.toHexString(),
+    event.block.timestamp,
+  );
+
+  const split = new Split(event.transaction.hash.toHexString());
+  split.timestamp = event.block.timestamp;
+  split.stakeholder = event.params.stakeholder.toHexString();
+  split.collateralToken = '{{lowercase contracts.USDC.address}}';
+  split.parentCollectionId = ByteArray.fromI32(0);
+  split.condition = event.params.conditionId.toHexString();
+  split.partition = [BigInt.fromI32(1), BigInt.fromI32(2)];
+  split.amount = event.params.amount;
+
+  split.save();
+}
+
+export function handlePositionsMerge(event: PositionsMerge): void {
+  requireAccount(event.params.stakeholder.toHexString(), event.block.timestamp);
+  markAccountAsSeen(
+    event.params.stakeholder.toHexString(),
+    event.block.timestamp,
+  );
+
+  const merge = new Merge(event.transaction.hash.toHexString());
+  merge.timestamp = event.block.timestamp;
+  merge.stakeholder = event.params.stakeholder.toHexString();
+  merge.collateralToken = '{{lowercase contracts.USDC.address}}';
+  merge.parentCollectionId = ByteArray.fromI32(0);
+  merge.condition = event.params.conditionId.toHexString();
+  merge.partition = [BigInt.fromI32(1), BigInt.fromI32(2)];
+  merge.amount = event.params.amount;
+
+  merge.save();
+}
+
+export function handlePositionsConverted(event: PositionsConverted): void {
+  requireAccount(event.params.stakeholder.toHexString(), event.block.timestamp);
+  markAccountAsSeen(
+    event.params.stakeholder.toHexString(),
+    event.block.timestamp,
+  );
+
+  const negRiskEvent = NegRiskEvent.load(event.params.marketId.toHexString());
+
+  if (negRiskEvent === null) {
+    return;
+  }
+
+  const conversion = new NegRiskConversion(
+    event.transaction.hash.toHexString(),
```

Comment:
Is it possible to have multiple conversions in the same transaction? If so, might be better to key conversions with transactionHash+logIndex